dashboards/hf_dataset_review.py

```diff
@@ -0,0 +1,37 @@
+import streamlit as st
+from datasets import load_dataset
+import streamlit_pianoroll
+import fortepyan as ff
+from object_generators.tokenizer_generator import TokenizerGenerator
+
+import json
+
+
+def main():
+    dataset_names = ["OneTimeTokDataset"]
+    dataset_name = st.selectbox(label="dataset", options=dataset_names)
+    dataset_split = st.selectbox(label="split", options=["train", "test", "validation"])
+    
+    if dataset_name == "OneTimeTokDataset":
+        dataset = load_dataset("./OneTimeTokDataset", name="debugging", split=dataset_split)
+    
+    idx = st.number_input(label="record_id", value=0, max_value=len(dataset))
+    record = dataset[idx]
+    
+    with st.expander(label="source"):
+        st.json(record["source"])
+        
+    tokenzier_generator = TokenizerGenerator()
+    tokenier_info = json.loads(dataset.description)
+    tokenizer = tokenzier_generator.generate_tokenizer(
+        name=tokenier_info["tokenizer_name"], 
+        parameters=tokenier_info["tokenizer_parameters"],
+    )
```

Comment:
```suggestion
    tokenizer_info = {"name": "foo", "parameters": "bar"}
    tokenizer = TokenizerGenerator.generate_tokenizer(
        name=tokenier_info["tokenizer_name"], 
        parameters=tokenier_info["tokenizer_parameters"],
    )
```

I don't think it's a good design to store tokenizer config as dataset description - is there any reason not to manage them separately?

Reply:
I cannot find any other way of storing custom metadata in huggingface datasets...

Reply:
I tried making a TokenizedDatasetInfo class (child of DatasetInfo) with additional field, but it throws error if it is not loaded from cache.

Reply:
I still don't see a reason to force the two to be together - I don't think we want to share on huggingface a general dataset builder supporting all the possible tokenizers 🤔

Reply:
I think treating tokenizer arguments just like other dataset arguments (sequence length, extra datasets etc.) would be better - it this possible here? 

Maybe the tokenizer-dataset relation should be well defined (only one tokenizer allowed), then you could then get rid of the tokenizer generator class that I don't like, and the code will become more explicit :)

Reply:
I was going for well-defined HF-dataset - tokenizer relation, while TorchDatasets define the specifc task (like next-token-prediction). BTW, I am also thinking about not uploading the builder for tokenized data to huggingface at all - and using them as a tool for handling our datasets inside this project. (in train.py you can see how that works locally)

Reply:
> I was going for well-defined HF-dataset - tokenizer relation

If that's the case, why do you store the tokenizer class name as a string, and why tokenizer parameters are stored in a dict? To me it's an antipattern and a sign that the design is broken 🤔

Reply:
Now that I think about it - we may also pass dataset_cfg as argument when initializing MidiDataset(TorchDataset) - and load the tokenized dataset during inizialization. That way we would not need to keep tokenizer information in the hf dataset.

Reply:
I was storing it this way because the custom MidiDataset(TorchDataset) needs information on which tokenizer was used to tokenize the dataset.
MidiDataset(TorchDataset) can use multiple different TokenizedDatasets(HuggingFaceDataset) for specific task.

Reply:
I think this is a clear indication that we're doing something wrong here. Please focus on the doc describing the data flow through pretokenizers and tokenizers that I asked for in another comment and we'll try to build from there.